students/GuyCChan/session08/circleClass.py

```diff
@@ -0,0 +1,117 @@
+from math import pi
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+
+# Chris, if I already have the properties decorator for diameter. What extra does
+# it do or what differences are there in making a classmethod for diameter?
+    @classmethod
+    def from_diameter(cls, diameter):
+        return cls(diameter / 2)
+
+    @property
+    def diameter(self):
+        """ Add properties decorator for diameter."""
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter_set(self, value):
+        """ Make properties setter for diameter."""
+        self.radius = value/2
+
+    @property
+    def area(self):
+        """ Find area using radius."""
+        return pi * self.radius**2
+
+# classmethod to find area using radius. I still haven't figured out the difference between this and the last"""
+# If this is a classmethod, then the first parameter should be 'cls' instead of 'self'. Please explain."""
+# This should only be necessary if the above method was not included at initialization of the class but is """
+# added later. Is this correct? """
+#     @classmethod
```

Comment:
The point of the "from_diameter" classmethod is to provide a way to create a new circle when you have the diameter. You don't need a "from_radius" because the regular initializer already uses the radius to make the Circle.